### PR TITLE
fix bug: custom prompt templates registered are never applied to vllm provider

### PR DESCRIPTION
**Pull request by zdsfwy**
### Issue Description

TLDR: custom prompt templates registered for local models are never applied to vllm provider.

I was trying to use litellm to build an openai compatible API on top of vllm, with local pre-downloaded model files. The problem here is that litellm would eventually falls back to `default_pt` method in prompt_template/factory.py because the model name interpreted by litellm is `/path/to/my/model/directory`. Given my code:
```python
response = litellm.completion(
            model="vllm//path/to/my/model/directory",
            ...
```
and registering custom prompt template for model `/path/to/my/model/directory` doesn't work because parameter `custom_prompt_dict` never passed to `vllm.completion` as shown below
```python
# code snippet from litellm/main.py
elif custom_llm_provider == "vllm":
            model_response = vllm.completion(
                model=model,
                messages=messages,
                model_response=model_response,
                print_verbose=print_verbose,
                optional_params=optional_params,
                litellm_params=litellm_params,
                logger_fn=logger_fn,
                encoding=encoding,
                logging_obj=logging,
            )
```
and code used to do the register:
```python
litellm.register_prompt_template(
        model="vllm//path/to/my/model/directory",
        initial_prompt_value="<s>",
        roles={
            "system": {
                "pre_message": "[INST] ",
                "post_message": " [/INST]",
            },
            "user": {"pre_message": "[INST] ", "post_message": " [/INST]"},
            "assistant": {"pre_message": "", "post_message": "</s> "},
        }
)
```

### CODE REVIEW

**Comment by ghost** [comment]
@zdsfwy can you share a screenshot of the code change working for you?

**Comment by zdsfwy** [comment]
Sure @krrishdholakia , here is the evidence. 
Before the fix, the prompt sent in request was `prompt='who are you'`(even though with custom prompt template registered):
image
After the fix, the prompt is `prompt='<s>[INST] who are you [/INST]'`
image
To test the fix, I reinstall litellm with a custom wheel built from my forked repo here: https://github.com/zdsfwy/litellm/releases/tag/v1.32.1-fix
And here is the code used for testing:
```python
import litellm 
litellm.set_verbose=True
litellm.vllm.llm = LLM(model=model_path, dtype="half")
litellm.register_prompt_template(
        model=f"vllm/{model_path}",
        initial_prompt_value="<s>",
        roles={
            "system": {
                "pre_message": "[INST] ",
                "post_message": " [/INST]",
            },
            "user": {"pre_message": "[INST] ", "post_message": " [/INST]"},
            "assistant": {"pre_message": "", "post_message": "</s> "},
        }
)
messages = [{"role": "user", "content": "who are you"}]
response = litellm.completion(
            model=f"vllm/{model_path}", # add a vllm prefix so litellm knows the custom_llm_provider==vllm
            messages=messages,
            temperature=0.2,
            max_tokens=80)
```

**Comment by zdsfwy** [comment]
Hi @krrishdholakia , please let me know whether anything else was need.

**Comment by ghost** [comment]
LGTM!
